filebeat/beater/channels.go

```diff
@@ -37,10 +40,16 @@ func newSpoolerOutlet(
 		done:    done,
 		spooler: s,
 		wg:      wg,
+		isOpen:  1,
 	}
 }
 
 func (o *spoolerOutlet) OnEvent(event *input.Event) bool {
+	open := atomic.LoadInt32(&o.isOpen) == 1
```

Comment:
I think this is ok and part of non-deterministic nature of spoolerOutlet. But one prospector go-routine will re-use the same channel. That is, while there is a chance some events still being published during close, the same prospector will not be able to publish events after having seen the close signal. Most important is the individual prospectors not publishing after having seen the channel being closed (directly or indirectly).